src/app/infrastructure/app-routing.module.ts

```diff
@@ -13,7 +13,7 @@ const routes: Routes = [
   {path: 'login', component: LoginComponent},
   {path: 'register', component: RegisterComponent},
   {path: 'unit/:unitId', component: UnitComponent, resolve : {unit: UnitResolver}},
-  {path: 'kc/:kcId', component: KnowledgeComponentComponent, canActivate: [AuthGuard]},
+  {path: 'kc/:kcId/learner/:learnerId', component: KnowledgeComponentComponent, canActivate: [AuthGuard]},
```

Comment:
We can't store the user Id in the URL. URLs are supposed to target specific resources. The user's id is an application-wide piece of data, similar to the selected theme. We already have a service for caching it. See https://github.com/Clean-CaDET/platform-tutor-ui-web/blob/master/src/app/modules/domain/feedback/feedback.service.ts#L21 for an example.